privx_api/privx_api.py

```diff
@@ -166,29 +166,18 @@ def _get_headers(self) -> dict:
             "Authorization": "Bearer {}".format(self._access_token),
         }
 
-    def _http_get(self, urlname: str) -> http.client.HTTPResponse:
-        conn = self._get_connection()
-        conn.request(
-            "GET",
-            self._get_url(urlname),
-            headers=self._get_headers(),
-        )
-
-        return conn.getresponse()
-
-    def _http_get_with_param(self, urlname: str, elem_id: str) -> http.client.HTTPResponse:
-
+    def _http_get(self, urlname: str, params: dict = {}) -> http.client.HTTPResponse:
         conn = self._get_connection()
+        url = self._get_url(urlname).format(**params)
```

Comment:
This doesn't allow passing query parameters, only path parameters.

Reply:
query params are in URL definition:
"rolestore.users.search": "/host-store/api/v1/hosts/search?offset={offset}&limit={limit}&sortkey={sortkey}&filter={filter}",

For path params:
 "rolestore.user.roles": "/role-store/api/v1/users/{user_id}/roles",

Reply:
That wouldn't work generally, as url parameters need to be url encoded. Also I don't think query parameters should be in that url definition.